common/thorhelper/thorsoapcall.cpp

```diff
@@ -2523,19 +2549,28 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                             idx = 0;
                         if (idx==startidx)
                         {
+                            master->activitySpanScope->recordException(e, true, true);
                             StringBuffer s;
                             master->logctx.CTXLOG("Exception %s", e->errorMessage(s).str());
                             processException(url, inputRows, e);
                             return;
                         }
                     } while (blacklist->blacklisted(url.port, url.host));
+
+                    master->activitySpanScope->recordException(e, false, false); //Record the exception, but don't set failure
                 }
             }
             try
             {
                 checkTimeLimitExceeded(&remainingMS);
                 checkRoxieAbortMonitor(master->roxieAbortMonitor);
+
+                StringBuffer spanName("SoapCall Socket Operation - ");
```

Comment:
Not convinced about this name.  I suspect it should not contain the url.  I'm not 100% sure what it should be though.

Reply:
Ok, you're not giving me much to go on. Changing it to Socket Write, which is 100% accurate, but in my opinion not as helpful. If that name doesn't convince you either, we'll have to discuss face to face and work out a name